blob/fileblob/fileblob.go

```diff
@@ -163,6 +164,9 @@ func escapeKey(s string) string {
 		// Escape the trailing slash in a key.
 		case c == '/' && i == len(r)-1:
 			return true
+		// https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file
+		case os.PathSeparator == '\\' && (c == '>' || c == '<' || c == ':' || c == '"' || c == '|' || c == '?' || c == '*'):
```

Comment:
Are you sure `os.PathSeparator` is a valid proxy for Windows? You could check `runtime.GOOS`.

Reply:
Windows definitely uses `\`, and I don't think any other operating systems do. I could be wrong.

`os.PathSeparator` is used in enough places in this file that I'd rather it be consistent. If there's an OS that uses `\` other than Windows, or an OS that uses a pathSeparator other than `/` or `\`, I'd be shocked if they worked out of the box regardless.